packages/snaps-rpc-methods/src/permitted/scheduleBackgroundEvent.ts

```diff
@@ -0,0 +1,154 @@
+import type { JsonRpcEngineEndCallback } from '@metamask/json-rpc-engine';
+import type { PermittedHandlerExport } from '@metamask/permission-controller';
+import { rpcErrors } from '@metamask/rpc-errors';
+import type {
+  JsonRpcRequest,
+  ScheduleBackgroundEventParams,
+  ScheduleBackgroundEventResult,
+} from '@metamask/snaps-sdk';
+import type { CronjobRpcRequest } from '@metamask/snaps-utils';
+import {
+  CronjobRpcRequestStruct,
+  type InferMatching,
+} from '@metamask/snaps-utils';
+import {
+  StructError,
+  create,
+  object,
+  refine,
+  string,
+} from '@metamask/superstruct';
+import { type PendingJsonRpcResponse } from '@metamask/utils';
+import { DateTime } from 'luxon';
+
+import { SnapEndowments } from '../endowments';
+import type { MethodHooksObject } from '../utils';
+
+const methodName = 'snap_scheduleBackgroundEvent';
+
+const hookNames: MethodHooksObject<ScheduleBackgroundEventMethodHooks> = {
+  scheduleBackgroundEvent: true,
+  hasPermission: true,
+};
+
+type ScheduleBackgroundEventHookParams = {
+  date: string;
+  request: CronjobRpcRequest;
+};
+
+export type ScheduleBackgroundEventMethodHooks = {
+  scheduleBackgroundEvent: (
+    snapEvent: ScheduleBackgroundEventHookParams,
+  ) => string;
+
+  hasPermission: (permissionName: string) => boolean;
+};
+
+export const scheduleBackgroundEventHandler: PermittedHandlerExport<
+  ScheduleBackgroundEventMethodHooks,
+  ScheduleBackgroundEventParameters,
+  ScheduleBackgroundEventResult
+> = {
+  methodNames: [methodName],
+  implementation: getScheduleBackgroundEventImplementation,
+  hookNames,
+};
+
+const ScheduleBackgroundEventsParametersStruct = object({
+  date: refine(string(), 'date', (val) => {
+    const date = DateTime.fromISO(val);
+    if (date.isValid) {
+      // luxon doesn't have a reliable way to check if timezone info was not provided
+      const count = val.split('').filter((char) => char === '-').length;
+      if (count !== 3 && !val.includes('+') && !val.endsWith('Z')) {
```

Comment:
This seems really complicated, I wonder if there's a simpler way to solve this problem 🤔

Reply:
> Perhaps we can allow omitting the timezone, but make a decision about what specific timezone to use if none is specified? As proposed by [MetaMask/SIPs#152](https://github.com/MetaMask/SIPs/pull/152)
> 
> cc @Mrtenz

The above logic would still have to be used to figure out if the timezone was omitted.

Reply:
Seems cleaner (and more performant?) to use a regular expression here, like `/[+-]\d{2}:?\d{2}|Z/`.

Reply:
We don't do what? Use regular expressions? I wouldn't say that, for certain things we definitely do 😄

Reply:
Haha ok, I'll simplify then.